packages/devextreme-react-generator/src/component-generator.test.ts

```diff
@@ -74,17 +74,79 @@ export {
   ).toBe(EXPECTED);
 });
 
+describe('generic types clause', () => {
+  it('is generated', () => {
+    // #region EXPECTED
+    const EXPECTED = `
+import dxCLASS_NAME, {
+    Properties
+} from "DX/WIDGET/PATH";
+
+import { Component as BaseComponent, IHtmlOptions } from "BASE_COMPONENT_PATH";
+
+interface ICLASS_NAMEOptions<T1 = any, T2 = any> extends Properties<T1, T2>, IHtmlOptions {
+  dataSource?: Properties<T1, T2>["dataSource"];
+}
+
+class CLASS_NAME<T1, T2> extends BaseComponent<ICLASS_NAMEOptions<T1, T2>> {
+
+  public get instance(): dxCLASS_NAME {
+    return this._instance;
+  }
+
+  protected _WidgetClass = dxCLASS_NAME;
+}
+export default CLASS_NAME;
+export {
+  CLASS_NAME,
+  ICLASS_NAMEOptions
+};
+`.trimLeft();
+      // #endregion
+
+    expect(
+      generate({
+        name: 'CLASS_NAME',
+        baseComponentPath: 'BASE_COMPONENT_PATH',
+        extensionComponentPath: 'EXTENSION_COMPONENT_PATH',
+        dxExportPath: 'DX/WIDGET/PATH',
+        optionsTypeParams: ['T1', 'T2'],
+      }),
+    ).toBe(EXPECTED);
+  });
+
+  it('is not generated for empty parrams array', () => {
+    expect(
+      generate({
+        name: 'CLASS_NAME',
+        baseComponentPath: 'BASE_COMPONENT_PATH',
+        extensionComponentPath: 'EXTENSION_COMPONENT_PATH',
+        dxExportPath: 'DX/WIDGET/PATH',
+        optionsTypeParams: [],
+      }),
+    ).toBe(
```

Comment:
It's unclear for me why this test is comparing two equivalent objects. Maybe new expected const is needed?